c7n/ufuncs/s3crypt.py

```diff
@@ -62,17 +62,7 @@ def process_key_event(event, context):
             # Ensure we know which key caused an issue
             print("error %s:%s code:%s" % (
                 bucket, key['Key'], e.response['Error']))
-            print("trying with urldecoded key")
-            try:
-                decode = urllib.unquote(key['Key'])
-                key['Key'] = decode
-                result = retry(method, s3, key, bucket)
-            except ClientError as e:
-                # Ensure we know which key caused an issue
-                print("failed with urldecoded key")
-                print("error %s:%s code:%s" % (
-                    bucket, key['Key'], e.response['Error']))
-                raise
+            raise
```

Comment:
raising causes a re-execution of the lambda, we need a dead letter queue possibly on these to stop the retry loops.

Reply:
Agreed - I'll add an issue to consider this.

Reply:
Added https://github.com/capitalone/cloud-custodian/issues/1569.